napari/layers/base/base.py

```diff
@@ -446,8 +446,8 @@ def affine(self):
 
     @affine.setter
     def affine(self, affine):
-        self._transforms['physical2world'] = coerce_affine(
-            affine, self.ndim, name='physical2world'
+        self._transforms[2] = coerce_affine(
```

Comment:
Definitely a fine fix for now, but something that long term should probably be fixed. Do you want to maybe add a comment here @andy-sweet saying something like "we have to use numerical index instead of name because ..." just so we remember that we don't love this pattern, but the alternative is worse.

Reply:
Done. Also wrote up #3058 and referenced that.